### ISSUE TITLE
reaper doesnt work correctly when to use backend postgre

### ISSUE THREAD

**Issue by cglrdev** [thelastpickle/cassandra-reaper#1019]
Hi,
when i install reaper(one reaper instance connect cluster,not sidecar mode) on memory backend good working but  backend postgre doesnt work correctly.
I add cluster via seed node only take seed node status. When i click the another nodes status button ,reaper get 500 error.
"no host could be reached through jmx" 
And not start repairing when i click the cluster repair button. also the web UI shows blank screen when i refresh the web site

i don't get this error when i use backend memory.

Ubuntu Version:20.04.1 LTS
Java 8
Postgresql version: 13.2
JMX auth cassandra integrated and basic auth tested ,same error

reaper config:

segmentCountPerNode: 64
repairParallelism: DATACENTER_AWARE
repairIntensity: 0.9
scheduleDaysBetween: 7
repairRunThreadCount: 15
hangingRepairTimeoutMins: 30
storageType: postgres
enableCrossOrigin: true
incrementalRepair: false
blacklistTwcsTables: true
enableDynamicSeedList: true
repairManagerSchedulingIntervalSeconds: 10
activateQueryLogger: false
jmxConnectionTimeoutInSeconds: 5
useAddressTranslator: false
#purgeRecordsAfterInDays: 30
#numberOfRunsToKeepPerUnit: 10

datacenterAvailability: LOCAL (ALL,EACH tested same error)

#jmxPorts:
#127.0.0.1: 7100
#127.0.0.2: 7200
#127.0.0.3: 7300
#127.0.0.4: 7400
#127.0.0.5: 7500
#127.0.0.6: 7600

#jmxAuth:
#username: cassandra
#password: cassandra

server:
  type: default
  applicationConnectors:
    - type: http
      port: 8080
      bindHost: 0.0.0.0
  adminConnectors:
    - type: http
      port: 8081
      bindHost: 0.0.0.0
  requestLog:
    appenders: []

postgres:
#driverClass: org.postgresql.Driver
  user: postgres
  password: postgresql123
  url: jdbc:postgresql://*.*.*.*:5432/reaperdb

cryptograph:
  type: symmetric
  systemPropertySecret: key

accessControl:
  sessionTimeout: PT10M
  shiro:
    iniConfigs: ["classpath:shiro.ini"]

**Comment by cglrdev** [comment]
I tested h2 backend today, same error

**Comment by adejanovski** [comment]
Hi, 

did you open your JMX ports to external access?
By default, Cassandra restricts it to localhost, which is only compatible with the SIDECAR mode.
You'll find some information on how to allow remote JMX access [here](https://docs.datastax.com/en/cassandra-oss/3.0/cassandra/configuration/secureJmxAuthentication.html).

**Comment by cglrdev** [comment]
Hi
I opened JMX external access.i don't get this error when i use backend memory.
sample config ips etc...
reaper  ip : *.*.*.20
cluster node 1: *.*.*.19 seed
cluster node 2:  *.*.*.18

**Comment by cglrdev** [comment]
I'm attached  log. also i tested all nodes jmx remote access jconsole success

[reaper.log](https://github.com/thelastpickle/cassandra-reaper/files/6023165/reaper.log)

**Comment by adejanovski** [comment]
Apparently you've enabled JMX remote access indeed but by default it requires authentication:

```
INFO   [2021-02-22 15:43:17,431] [ClusterResource] i.c.j.JmxConnectionFactory - Unreachable host: Failure when establishing JMX connection to *.*.*.19:7199: java.lang.SecurityException: Authentication failed! Credentials required 
ERROR  [2021-02-22 15:43:47,500] [ClusterResource] i.c.j.JmxConnectionFactory - Failed creating a new JMX connection to *.*.*.19:7199 
```

Did you register the cluster in Reaper with the right credentials? Are these credentials the same on all nodes?

**Comment by cglrdev** [comment]
Yes.I tested on  jconsole.
also i add cluster via non seed node(19 ip) then success  get  node status :) (but my workaround doesnt work in a another 6 node cluster, 2 node cluster working)
so there is no problem  jmx external access (same user pw)
as i said i working correctly with backend  storage type memory ( all nodes status working )

**Comment by burakbalim** [comment]
This issue solved. But I couldn't push commit. ClusterMapper on line 63. We need to add withJmxCredentials

  Cluster.Builder builder = Cluster.builder()
            .withName(rs.getString("name"))
            .withSeedHosts(ImmutableSet.copyOf(seedHosts))
            .withState(null != rs.getString("state")
                    ? Cluster.State.valueOf(rs.getString("state"))
                    : Cluster.State.UNKNOWN)
            .withLastContact(lastContact)
            **.withJmxCredentials(clusterProperties.getJmxCredentials())**
            .withJmxPort(clusterProperties.getJmxPort());

**Comment by adejanovski** [comment]
Oh wow, that's no good indeed 😅 . I'll create a PR for this.

**Comment by cglrdev** [comment]
Thanks @burakbalim @adejanovski 

when do you release fix? @adejanovski

**Comment by adejanovski** [comment]
Most probably tomorrow. I have another fix to push on the Cassandra backend.

**Comment by adejanovski** [comment]
I just released 2.2.1 with this fix
